packages/js/src/website.ts

```diff
@@ -61,45 +65,94 @@ type FormbricksWebsiteMethods = {
 const formbricksProxyHandler: ProxyHandler<TFormbricksWebsite> = {
   get(_target, prop, _receiver) {
     return async (...args: any[]) => {
-      if (!window.formbricks) {
-        if (prop !== "init") {
-          console.error(
-            "🧱 Formbricks - Global error: You need to call formbricks.init before calling any other method"
-          );
-          return;
-        }
+      const methodQueue = new MethodQueue();
 
-        // still need to check if the apiHost is passed
-        if (!args[0]) {
-          console.error("🧱 Formbricks - Global error: You need to pass the apiHost as the first argument");
+      if (prop === "init") {
+        // if still initializing, return
+        if (isInitializing) {
```

Comment:
The js-core in this case also throws a warning that it's already initialized. I think I would like to keep this and let js-core handle this case.

So we decide the cases different on the top level here:

- if !initialized
  - if initializing
     - add to queue
   - else:
     - it's an init call -> initialize + process queue
     - else: error "call init first"
 - else:
    - add to queue + process queue